assembly/src/ast/nodes/serde/debug.rs

```diff
@@ -0,0 +1,32 @@
+use super::{super::DebugOptions, ByteReader, ByteWriter, DeserializationError, ToString};
+
+const STACK_ALL: u8 = 0;
+const STACK_TOP: u8 = 1;
+
+/// Writes the provided [DebugOptions] into the provided target.
+pub fn write_options_into<W: ByteWriter>(target: &mut W, options: &DebugOptions) {
+    match options {
+        DebugOptions::StackAll => target.write_u8(STACK_ALL),
+        DebugOptions::StackTop(n) => {
+            target.write_u8(STACK_TOP);
+            target.write_u8(*n);
+        }
+    }
+}
+
+/// Reads [DebugOptions] from the provided source.
+pub fn read_options_from<R: ByteReader>(
+    source: &mut R,
+) -> Result<DebugOptions, DeserializationError> {
+    match source.read_u8()? {
+        STACK_ALL => Ok(DebugOptions::StackAll),
+        STACK_TOP => {
+            let n = source.read_u8()?;
+            if n == 0 {
+                return Err(DeserializationError::InvalidValue(n.to_string()));
+            }
```

Comment:
Do we need to apply an upper limit on the value of `n` or is 255 (`u8::MAX`) valid?  If 255 is valid then is a higher number also valid?  As I understand it, the stack can reach a depth of `2^32`.

Reply:
We need to apply some upper limit - but yeah, 255 is probably too restrictive. I changed this to `u16`. I think this should be sufficient for most cases, and in cases when the user needs to print out more, they can use `stack.all`.